src/main/java/com/google/jenkins/plugins/k8sengine/KubernetesVerifiers.java

```diff
@@ -0,0 +1,198 @@
+    private static final String AVAILABLE_REPLICAS = "availableReplicas";
+    private static final String MINIMUM_REPLICAS_JSONPATH = "spec.replicas";
+    private static final String STATUS_JSONPATH = "status";
+
+    public VerificationResult verify(KubectlWrapper kubectl, Manifests.ManifestObject object) {
+      LOGGER.info(String.format("Verifying deployment, %s", object.getName()));
+      StringBuilder log = new StringBuilder();
+      Object json = null;
+
+      try {
+        json = kubectl.getObject(object.getKind().toLowerCase(), object.getName());
+      } catch (Exception e) {
+        return errorResult(e, object);
+      }
+
+      Integer minReplicas = JsonPath.read(json, MINIMUM_REPLICAS_JSONPATH);
+      Map<String, Object> status = JsonPath.read(json, STATUS_JSONPATH);
+      Integer availableReplicas = (Integer) status.getOrDefault(AVAILABLE_REPLICAS, 0);
+      boolean verified =
+          minReplicas != null
+              && availableReplicas != null
+              && minReplicas.intValue() <= availableReplicas.intValue();
+
+      log.append("AvailableReplicas = ")
+          .append(availableReplicas)
+          .append(",")
+          .append(" MinimumReplicas = ")
+          .append(minReplicas)
+          .append("\n");
+
+      return new VerificationResult(log.toString(), verified, object);
+    }
+  }
+
+  private static final Verifier defaultVerifier = new DefaultVerifier();
+
+  /*
+   * register verifiers
+   */
+  static {
+    verifiers.put(kindIdentifier("deployment"), new DeploymentVerifier());
+  }
+  /*
+   * create an identifier for looking up a verifier for an object kind for any apiVersion
+   */
+  private static String kindIdentifier(String kind) {
+    return "*/" + kind.toLowerCase();
+  }
+
+  private static Verifier getKindVerifier(String kind) {
+    return verifiers.get(kindIdentifier(kind));
+  }
+  /**
+   * Return the verifier for the api object.
+   *
+   * @param apiVersion The kubernetes api version.
+   * @param kind The kubernetes object kind.
+   */
+  private static Verifier getVerifier(String apiVersion, String kind) {
+    Verifier verifier = verifiers.get(apiVersion + "/" + kind);
+    if (verifier == null) {
+      verifier = getKindVerifier(kind);
+      if (verifier == null) {
+        verifier = defaultVerifier;
+      }
+    }
+    return verifier;
+  }
+
+  /**
+   * @param kubectl The kubectl wrapper that will query the cluster.
+   * @param object The representation of the kubernetes object to verify.
+   * @return VerificationResult Object that encapsulates whether the kubectl object was verified
+   *     together with relevant verification log that is dependent on the type of kubernetes object.
+   */
+  public static VerificationResult verify(KubectlWrapper kubectl, Manifests.ManifestObject object) {
+    if (object == null) {
+      return errorResult("No manifest object.", null);
+    }
+    Verifier verifier = getVerifier(object.getApiVersion(), object.getKind());
```

Comment:
These two lines can be condensed to: `return getVerifier(object.getApiVersion(), object.getKind()).verify(kubectl, object);`